From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron Friel <mayreply@aaronfriel.com>
Date: Mon, 6 Mar 2023 19:13:42 -0800
Subject: [PATCH] Add `shim.NewProvider` to expose `internal/provider.New`


diff --git a/shim/shim.go b/shim/shim.go
new file mode 100644
index 00000000..e6ecce26
--- /dev/null
+++ b/shim/shim.go
@@ -0,0 +1,17 @@
+package shim
+
+import (
+	pf "github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	pfProvider "github.com/aiven/terraform-provider-aiven/internal/plugin"
+	"github.com/aiven/terraform-provider-aiven/internal/sdkprovider/provider"
+)
+
+func NewProvider(version string) (*schema.Provider, error) {
+	return provider.Provider(version)
+}
+
+func NewPFProvider(version string) pf.Provider {
+	return pfProvider.New(version)()
+}
