From 6d9aa380c452d0481a0683c9e747a77775816c5a Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Wed, 22 Mar 2023 16:02:19 -0400
Subject: [PATCH] Avoid panic in custom_diff

---
 internal/schemautil/custom_diff.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/internal/schemautil/custom_diff.go b/internal/schemautil/custom_diff.go
index 3d6bf64..a57f299 100644
--- a/internal/schemautil/custom_diff.go
+++ b/internal/schemautil/custom_diff.go
@@ -220,6 +220,11 @@ func typedKeys(m map[string]cty.Value) map[string]struct{} {
 // CustomizeDiffDisallowMultipleManyToOneKeys checks that we don't have multiple keys that are going to be converted to
 // a single key in the API request, e.g. 'ip_filter' and 'ip_filter_object' in the same diff.
 func CustomizeDiffDisallowMultipleManyToOneKeys(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	// Avoid panic in d.GetRawConfig().AsValueMap()
+	if d.GetRawConfig().IsNull() {
+		return nil
+	}
+
 	ks := d.GetRawConfig().AsValueMap()
 
 	tks := map[string]struct{}{}
-- 
2.37.1 (Apple Git-137.1)

