From 7e63e0bf8f44415fcfad9c3f204cac4ae4b0b399 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Thu, 20 Apr 2023 16:24:38 -0400
Subject: [PATCH 2/3] Avoid panic in custom_diff

---
 internal/schemautil/custom_diff.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/internal/schemautil/custom_diff.go b/internal/schemautil/custom_diff.go
index 5019feb..ed27976 100644
--- a/internal/schemautil/custom_diff.go
+++ b/internal/schemautil/custom_diff.go
@@ -168,6 +168,10 @@ func CustomizeDiffCheckStaticIPDisassociation(_ context.Context, d *schema.Resou
 // CustomizeDiffDisallowMultipleManyToOneKeys checks that we don't have multiple keys that are going to be converted to
 // a single key in the API request, e.g. 'ip_filter' and 'ip_filter_object' in the same diff.
 func CustomizeDiffDisallowMultipleManyToOneKeys(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	// Avoid panic in d.GetRawConfig().AsValueMap()
+	if d.GetRawConfig().IsNull() {
+		return nil
+	}
 	for k, v := range d.GetRawConfig().AsValueMap() {
 		if strings.Contains(k, "_user_config") { // we only care about *_user_config
 			if err := checkForMultipleValues(v); err != nil {
-- 
2.39.3 (Apple Git-145)

