From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Kappler <tkappler@pulumi.com>
Date: Thu, 28 Sep 2023 20:48:51 +0200
Subject: [PATCH] Internal provider has moved, update shim


diff --git a/shim/shim.go b/shim/shim.go
index 796cd94a..9a54b694 100644
--- a/shim/shim.go
+++ b/shim/shim.go
@@ -4,7 +4,7 @@ import (
 	pf "github.com/hashicorp/terraform-plugin-framework/provider"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
-	pfProvider "github.com/aiven/terraform-provider-aiven/internal/provider"
+	pfProvider "github.com/aiven/terraform-provider-aiven/internal/plugin"
 	"github.com/aiven/terraform-provider-aiven/internal/sdkprovider/provider"
 )
 
